docs/service/s3-stream-wrapper.rst

```diff
@@ -23,6 +23,12 @@ This allows you to access buckets and objects stored in Amazon S3 using the
 bucket name followed by a forward slash and an optional object key or prefix:
 ``s3://<bucket>[/<key-or-prefix>]``.
 
+.. note::
+
+    The stream wrapper is designed for working with objects and buckets on which
+    you have at least read permission. It is recommended that you use S3 client
```

Comment:
I was thinking of [the S3 ACL 'read' grant](http://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html), which corresponds to `ListBucket`, `ListBucketVersions`, and `ListBucketMultipartUploads` for buckets and `GetObject`, `GetObjectVersion`, and `GetObjectTorrent` for objects. I'll call those out here, as that's a very specific definition of 'read.'